dev-utils/update-tav-versions.js

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+// Update version property for each module defined in .tav.yml file
+// Usage:
+//      node dev-utils/update-tav-versions.js
+
+const { exec } = require('child_process');
+const { readFileSync, writeFileSync } = require('fs');
+const path = require('path');
+
+const semver = require('semver');
+const yaml = require('js-yaml');
+
+const TOP = path.resolve(path.join(__dirname, '..'));
+const TAV_PATH = path.join(TOP, '.tav.yml');
+
+async function main() {
+  const tavContent = readFileSync(TAV_PATH, { encoding: 'utf-8' });
+  const tavConfig = yaml.load(tavContent);
+  const tavEntries = Object.entries(tavConfig);
+  const modNames = new Set();
+  const versionsMap = new Map();
+
+  const versionModes = {
+    'latest-minors': getLatestMinors,
```

Comment:
Did you intend to do "latest-majors" and "latest" (as mentioned in the issue description) in this PR? They could come in separate PRs later when/if we need them.

Reply:
modes are quite straightforward so I've added them all. They are separated in functions for better review